stingray/stats.py

```diff
@@ -0,0 +1,268 @@
+import warnings
+from collections.abc import Iterable
+
+import numpy as np
+from scipy import stats
+
+
+__all__ = ['p_multitrial_from_single_trial',
+           'p_single_trial_from_p_multitrial',
+           'fold_profile_probability',
+           'fold_detection_level',
+           'z2_n_detection_level',
+           'z2_n_probability']
+
+
+def p_multitrial_from_single_trial(p1, n):
+    """Calculate a multi-trial probability from a single-trial one.
+
+    Calling _p_ the probability of a single hit, the Binomial
+    distributions says that the probability _at least_ one outcome
+    in n trials is
+
+                         n
+    P (k ≥ 1) =   Σ    (   ) p^k (1 - p)^(n - k)
+                k ≥ 1    k
+
+    or more simply, using P(k ≥ 0) = 1
+                          n
+    P (k ≥ 1) =   1 -   (   ) (1 - p)^n = 1 - (1 - p)^n
+                          0
+    """
+    return 1 - (1 - np.longdouble(p1))**np.longdouble(n)
+
+
+def p_single_trial_from_p_multitrial(pn, n):
+    """Calculate a single-trial probability from a multi-trial one.
```

Comment:
I'm sorry, I don't really understand what this function does? Is this for correcting the signficance threshold only, or does this have some other usage?

Reply:
Let's say I want to set the threshold for a search, so that I have 1% chance of detecting a "pulsation" that is actually produced by noise. I have to take into account the number of "trials", the number of independent spectral bins, so that the 1% probability is really the probability of detecting something spurious in the _complete_ spectrum. 

We (all) do epsilon / ntrial for that. Right? We usually calculate the probability of noise powers reaching epsilon / ntrial. But we all also know that epsilon / ntrial is an approximation

This here is the correct formula, inverting the formula for the probability after many trials obtained through the binomial distribution.

Reply:
This is the Šidák correction, which is more precise than the Bonferroni correction. A substitute for epsilon/n